src/main/java/core/basesyntax/storage/Storage.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.storage;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public void put(String fruit, int quantity) {
+        storage.put(fruit, quantity);
+    }
+
+    public int get(String fruit) {
+        return storage.getOrDefault(fruit, 0);
```

`get(String fruit)` must validate the `fruit` parameter (non-null/non-blank) and throw an informative exception when invalid. The current implementation returns a default for null which may hide incorrect calls; the checklist expects explicit validation.